src/main/java/ca/waterloo/dsg/graphflow/query/structuredquery/QueryPropertyInClause.java

```
@@ -0,0 +1,48 @@
+package ca.waterloo.dsg.graphflow.query.structuredquery;
+
+import ca.waterloo.dsg.graphflow.util.JsonOutputable;
+import com.google.gson.JsonObject;
+import org.antlr.v4.runtime.misc.Pair;
+
+import java.util.ArrayList;
+
+/**
+ * Created by naren on 9/22/17.
+ */
+public class QueryPropertyInClause implements JsonOutputable {
+
+    private Pair<String, String> leftOperand;
+    private ArrayList<Pair<String, String>> variableList;
+    private ArrayList<String> literalList;
+
+    public QueryPropertyInClause() { }
+
+    public Pair<String, String> getLeftOperand() {
+        return leftOperand;
+    }
+
+    public void setLeftOperand(Pair<String, String> leftOperand) {
+        this.leftOperand = leftOperand;
+    }
+
+    public ArrayList<Pair<String, String>> getVariableList() {
+        return variableList;
+    }
+
+    public void setVariableList(ArrayList<Pair<String, String>> variableList) {
+        this.variableList = variableList;
+    }
+
+    public ArrayList<String> getLiteralList() {
+        return literalList;
+    }
+
+    public void setLiteralList(ArrayList<String> literalList) {
+        this.literalList = literalList;
+    }
+
+    @Override
+    public JsonObject toJson() {
```

Make sure to implement this as it is required for logs but also by the plan viewer.
Generally until something is implemented, it should throw an UnsupportedOperationException.
The InClauses toJson() should be called in toJson() method of query/operator/filter/Filter.java